src/raft/raft.rs

```diff
@@ -1711,6 +1752,20 @@ impl<T: Storage> Raft<T> {
             return Some(false);
         }
 
+        if !self.is_learner {
```

Comment:
I don't think this is going to work. When a raft node is created, it's not learner by default. So there is no way for a new learner to accept snapshot. So should either set different flag for learner messages or also check if the state machine is initialized.

Reply:
I think we can pass learners info in `Config`? So the enw learner can set `self.is_learner = true` in `Raft::new`.

Reply:
When a raft state machine is created via conf change, there is no way to tell if it's a learner or a voter in current implementations.